web/chromecom.js

```diff
@@ -300,7 +300,7 @@ function setReferer(url, callback) {
 let storageArea = chrome.storage.sync || chrome.storage.local;
 
 class ChromePreferences extends BasePreferences {
-  _writeToStorage(prefObj) {
+  async _writeToStorage(prefObj) {
```

Comment:
Now that this is `async`, we should be able to remove the `return new Promise` code below. This also applies to the change just below and the changes in `web/firefoxcom.js`.

Reply:
> Now that this is async, we should be able to remove the return new Promise code a line below.

Nope, that won't work :-)
Please note that, both here and in the other file you refer to, the `Promise`s are resolved from callback functions and that doesn't really work with `async`/`await`. (Maybe these shouldn't be marked as `async`, but that'd look slightly strange since the `BasePreferences` ones are.)

---

> I really like how this works and how it makes the code easier. 

It can (greatly) simplify certain code, as long as you keep in mind that not all `Promise` usage can/should be replaced with `async`/`await`; it really depends on the situation!
Keep in mind that `await` will *suspend* the following code, whereas a `Promise` allows later code to run; [these examples](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#Examples) are quite good at highlighting the differences in behaviour.